boards/beagle/beagleconnect_freedom/board_antenna.c

```diff
@@ -1,116 +1,119 @@
-			RF_GlobalEventRadioPowerDown),
+	.globalEventMask = (RF_GlobalEventRadioSetup | RF_GlobalEventRadioPowerDown),
 };
 
+PINCTRL_DT_INST_DEFINE(0);
+DEVICE_DT_INST_DEFINE(0, board_antenna_init, NULL, NULL, NULL, POST_KERNEL,
+		      CONFIG_BOARD_ANTENNA_INIT_PRIO, NULL);
+
+#define GPIO_DT_SPEC(n, p, i) GPIO_DT_SPEC_GET_BY_IDX(n, p, i),
+
+static const struct pinctrl_dev_config *ant_pcfg = PINCTRL_DT_INST_DEV_CONFIG_GET(0);
+static const struct gpio_dt_spec ant_gpios[] = {
+	DT_FOREACH_PROP_ELEM(DT_NODELABEL(antenna_mux0), gpios, GPIO_DT_SPEC)};
+
 /**
  * Antenna switch GPIO init routine.
  */
-static int board_antenna_init(void)
+int board_antenna_init(const struct device *dev)
 {
+	ARG_UNUSED(dev);
+	int i;
 
-	/* set all paths to low */
-	IOCPinTypeGpioOutput(BOARD_RF_HIGH_PA);
-	GPIO_setOutputEnableDio(BOARD_RF_HIGH_PA, GPIO_OUTPUT_DISABLE);
-	IOCPinTypeGpioOutput(BOARD_RF_SUB1GHZ);
-	GPIO_setOutputEnableDio(BOARD_RF_SUB1GHZ, GPIO_OUTPUT_DISABLE);
+	/* default pinctrl configuration: set all antenna mux control pins as GPIOs */
+	pinctrl_apply_state(ant_pcfg, PINCTRL_STATE_DEFAULT);
+	/* set all GPIOs to 0 (all RF paths disabled) */
+	for (i = 0; i < ARRAY_SIZE(ant_gpios); i++) {
+		gpio_pin_configure_dt(&ant_gpios[i], 0);
+	}
 	return 0;
 }
 
-SYS_INIT(board_antenna_init, POST_KERNEL, CONFIG_BOARD_ANTENNA_INIT_PRIO);
-
+/**
+ * Custom TI RFCC26XX callback for switching the on-board antenna mux on radio setup.
+ */
 void board_cc13xx_rf_callback(RF_Handle client, RF_GlobalEvent events, void *arg)
 {
-	bool    sub1GHz   = false;
+	bool sub1GHz = false;
 	uint8_t loDivider = 0;
+	int i;
 
-	/* Switch off all paths first. Needs to be done anyway in every sub-case below. */
-	GPIO_setOutputEnableDio(BOARD_RF_HIGH_PA, GPIO_OUTPUT_DISABLE);
-	GPIO_setOutputEnableDio(BOARD_RF_SUB1GHZ, GPIO_OUTPUT_DISABLE);
+	/* Clear all antenna switch GPIOs (for all cases). */
+	for (i = 0; i < ARRAY_SIZE(ant_gpios); i++) {
+		gpio_pin_configure_dt(&ant_gpios[i], 0);
+	}
 
 	if (events & RF_GlobalEventRadioSetup) {
 		/* Decode the current PA configuration. */
-		RF_TxPowerTable_PAType paType = (RF_TxPowerTable_PAType)
-			RF_getTxPower(client).paType;
+		RF_TxPowerTable_PAType paType =
+			(RF_TxPowerTable_PAType)RF_getTxPower(client).paType;
 		/* Decode the generic argument as a setup command. */
 		RF_RadioSetup *setupCommand = (RF_RadioSetup *)arg;
 
 		switch (setupCommand->common.commandNo) {
-		case (CMD_RADIO_SETUP):
-		case (CMD_BLE5_RADIO_SETUP):
+		case CMD_RADIO_SETUP:
+		case CMD_BLE5_RADIO_SETUP:
 			loDivider = RF_LODIVIDER_MASK & setupCommand->common.loDivider;
-			/* Sub-1GHz front-end. */
-			if (loDivider != 0)
-				sub1GHz = true;
 			break;
-		case (CMD_PROP_RADIO_DIV_SETUP):
+		case CMD_PROP_RADIO_DIV_SETUP:
 			loDivider = RF_LODIVIDER_MASK & setupCommand->prop_div.loDivider;
-			/* Sub-1GHz front-end. */
-			if (loDivider != 0)
-				sub1GHz = true;
 			break;
 		default:
 			break;
 		}
+		sub1GHz = (loDivider != 0);
```

Comment:
This seems like a more convoluted approach than the original. Let the optimizer do it's job. As best I can tell, the result would be the same since both loDivider and sub1GHz are non-static and loDivider initializes to 0, but I think this is just harder to read for no performance gain.

Reply:
Again, this is not a change in original but a diff between 2 completely different files. Also, I can't say I prefer the old style but I do not have strong opinion either way.

Reply:
With this much similarity, they aren't completely different files. There's missing history here and that's "not good". We want Zephyr's history to reflect the reason changes were made.